File: config/config-sbc2ha/src/main/java/com/dfi/sbc2ha/config/sbc2ha/definition/sensor/ScalarSensorConfig.java
package com.dfi.sbc2ha.config.sbc2ha.definition.sensor;

import com.dfi.sbc2ha.config.sbc2ha.definition.filters.ValueFilterType;
import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

@Getter
@Setter
public abstract class ScalarSensorConfig extends SensorConfig {
    private List<Map<ValueFilterType, Number>> filters = new ArrayList<>();

    public ScalarSensorConfig() {
        super();
    }

}

File: config/editor/src/app/definition/sensor/ScalarSensorConfig.ts
import {ValueFilterType} from "../enums/ValueFilterType";
import {SensorConfig} from "./SensorConfig";
import {PlatformType} from "../enums/PlatformType";

export abstract class ScalarSensorConfig extends SensorConfig {
  filters: Filter[] = [];

  constructor(platform: PlatformType, name: string) {
    super(platform, name);
  }
}

export class Filter {
  type: ValueFilterType;
  value: number;

  constructor(type: ValueFilterType, value: number) {
    this.type = type;
    this.value = value;
  }
}
